src/common/mfu_io.c

```diff
@@ -326,6 +697,26 @@ int mfu_close(const char* file, int fd)
     return rc;
 }
 
+int mfu_file_close(const char* file, mfu_file_t* mfu_file)
+{
+    if (mfu_file->type == POSIX) {
+        int rc = mfu_close(file, mfu_file->fd);
+        return rc;
```

Comment:
if rc == 0 here, let's set mfu_file->fd back to its initial value (-1?), with a similar change for daos.  That might help use catch other bugs.  But only on success.  On error, let's leave it as it, because they may still need to do something with it in that case.